success/index.js

```diff
@@ -1,66 +1,210 @@
+      } catch (e) {
+        console.log('Error connecting to Host HP Admin URL :', e)
+        return null
+      }
+      // Once Host HP Admin URL returns status 200, progress to final page.
+      updateUiStep(6)
+    },
+    goToHPAdmin: () => {
+      /* Communicate visually that something is happening in the bkgd */
+      buttons.goToHPAdmin.disabled = true
+      setTimeout(() => {
+        /* Clean State */
+        buttons.goToHPAdmin.disabled = false
+      }, 500)
+    }
+  }
+
+  // /* Bind actions to buttons */
+  buttons.goToHPAdmin.onclick = actions.goToHPAdmin
+
+  /** Helper Functions :
+  * =============================
+  *
+  */
+  const validation = { 5: !0, 6: !0 }
+
+  const addMinutesToDateTime = (dt, minutes) => new Date(dt.getTime() + minutes * 60000)
+
+  /**
+  * Listen to stepTracker to initiate step specific actions
+  */
+  const constantCheck = () => {
+    if (stepTracker === 5) {
+      /* Start Timer */
+      const deadline = addMinutesToDateTime(new Date(), 30)
+
+      /* DEV HACK: exchange 30min for 10 sec timer during dev mode: */
+      // const deadline = addMinutesToDateTime(new Date(), 0.1)
+
+      console.log('Network Setup Deadline : ', deadline)
+      actions.countdownTimer(deadline)
+      actions.checkHpAdminState()
+    }
+  }
 
   /**
-   * Keep fetching given url until successful 200 response in 2s intervals
-   * @param {string} url 
-   * @returns {Promise}
+  * Resolve promise after specified time delay (in ms)
+  */
+  const delay = time => {
+    return new Promise(resolve => {
+      setTimeout(resolve, time)
+    })
+  }
+
+  /**
+   * Poll for Host's HP Admin Domain to determine network status
+   *
+   * @param {String} url
+   * @param {Int} interval (ie: Polling frequency (in ms). NB: 0 means try continuously.)
+   * @param {Int} timeout (ie: Polling duration (in ms).)
    */
-  function fetchRetry(url) {
+  const checkHpAdminUrl = (url, interval, timeout) => {
+    const start = Date.now()
     const fetchOptions = {
-      cache: 'no-cache'
+      cache: 'no-cache',
+      mode: 'no-cors'
     }
-    const delay = 2000
-
-    return new Promise((resolve,reject) => {
-        const success = (r) => {
-          if (r.status === 200) {
-            console.log('Fetch success!')
-            resolve(r)
-          } else {
-            console.error('Error in response status: ', r.status, ' retrying')
-            setTimeout(fetchUrl,delay)
-          }
-        }
-        const failure = (e) => {
-          console.error('Fetch error:', e)
-          setTimeout(fetchUrl,delay)
-        }
-        const fetchUrl = () => {
-            return fetch(url + 'ping/',fetchOptions)
-                .then(success)
-                .catch(failure)
+
+    if (timeout !== 0 && Date.now() - start > timeout) {
+      throw new Error('Timeout calculation error.')
+    }
+
+    return new Promise(resolve => {
+      const success = result => {
+        if (result.status === 200) {
```

Comment:
This won't work because of `CORB`.
Try to:
- open `https://example.com/`
- open dev tools
- in console type:
```javascript
fetch("https://www.google.com", {
      cache: 'no-cache',
      mode: 'no-cors'
}).then(r => {
	console.log('STATUS: ', r.status);
})
```
You will always get `STATUS: 0` because `CORB` is blocking result of request from reaching javascript. This needs to be some kind of call that is not subject to `CORB`. Let me know if you need help figuring out what exactly.

Reply:
For example we might wanna make a POST call to `url + 'api/v1/' and if we get `result.status === 401` then it means HP is reachible. But let me check if that is subject to `CORB` or not